src/credentials/FabricTable.cpp

```diff
@@ -787,8 +816,230 @@ void FabricTable::DeleteAllFabrics()
     }
 }
 
+const char * FabricTable::FormatChipEpochTime(System::Clock::Seconds32 chipEpochTime)
+{
+    static char buf[strlen("2000-00-00T00:00:00") + 1] = { 0 };
+    uint16_t year;
+    uint8_t month;
+    uint8_t day;
+    uint8_t hour;
+    uint8_t minute;
+    uint8_t second;
+    ChipEpochToCalendarTime(chipEpochTime.count(), year, month, day, hour, minute, second);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
```

Comment:
Please use ifdefs based on compiler here not to use pragmas against a compiler not supporting them.

Reply:
Alternatively, I could just make the buffer bigger to prevent the warnings.  Would that be better?

Reply:
In the updated code, this pragma is replaced with a runtime check to overcome the warning.